utils/extractor/src/index.ts

```diff
@@ -293,3 +293,53 @@ export type SheetCapture = {
   data: Flatfile.RecordData[]
   metadata?: { rowHeaders: number[] }
 }
+
+export function parseSheet(jsonArray: any[]): SheetCapture {
+  try {
+    // Ensure all items are objects
+    const filteredResults = jsonArray.filter(
+      (item) => typeof item === 'object' && item !== null
+    )
+
+    if (filteredResults.length === 0) {
+      return {} as SheetCapture
+    }
+
+    // Custom flatten function
+    const flattenObject = (obj: any, parent: string = '', res: any = {}) => {
+      for (let key in obj) {
+        const propName = parent ? parent + '.' + key : key
+        if (typeof obj[key] === 'object') {
+          flattenObject(obj[key], propName, res)
+        } else {
+          res[propName] = obj[key]
+        }
+      }
+      return res
+    }
```

Comment:
AI suggestions around type safety https://github.com/FlatFilers/flatfile-plugins/pull/693/files#r1826003574

Much of this doesn't appear to achieve anything significant. Ignoring the depth check (addressed above) usage of unknown, calling `hasOwnProperty` on keys being pulled from the object itself, and doing a null check before a type check don't add type safety as far as I can tell

The array check is interesting however, as I don't think the initial implementation handled that. Let me see how it behaves for arrays and I'll post back here

Reply:
Actually part of this Array edge case seems to be handled below when computing the filtered data via `value: Array.isArray(cell) ? JSON.stringify(cell) : cell,`. Will verify. I think it's worth adding some unit tests as well for parseSheet now that it's in the utils package

Reply:
Added unit tests that indicate these checks aren't necessary and/or already exist more succinctly elsewhere. Product approach here appears to be that array items are split into columns. E.g.
```json
"Nicknames": [
      "Dr. T",
      "Señior Lamb"
    ],
```
turns into these sheet rows:
```json
"Nicknames.0": { "value": "Dr. T" },
"Nicknames.1": { "value": "Señior Lamb" },
```

If an array is empty, the resulting column is null. E.g.
```
"Nicknames.0": { "value": undefined },
```

The test added to the `utils/extractor` package captures this